src/sgp/Types.cc

```diff
@@ -0,0 +1,140 @@
+#include "Types.h"
+
+#include "Campaign_Types.h"
+
+#define MAX_SECTOR 16
+#define MAX_Z 3
+#define MAP_WORLD_X 18 // redefine to avoid needing to include StrategicMap.h
+
+static UINT8 getXfromSectorID(UINT32 sectorId)
+{
+	return sectorId % MAX_SECTOR + 1;
+}
+
+static UINT8 getYfromSectorID(UINT32 sectorId)
+{
+	return sectorId / MAX_SECTOR + 1;
+}
+
+SGPSector::SGPSector(UINT32 s) noexcept
+{
+	x = getXfromSectorID(s);
+	y = getYfromSectorID(s);
+}
+
+static UINT16 getXfromStrategicIndex(UINT16 idx)
+{
+	return idx % MAP_WORLD_X;
+}
+
+static UINT16 getYfromStrategicIndex(UINT16 idx)
+{
+	return idx / MAP_WORLD_X;
+}
+
+SGPSector SGPSector::FromStrategicIndex(UINT16 idx)
+{
+	return SGPSector(getXfromStrategicIndex(idx), getYfromStrategicIndex(idx));
+}
+
+SGPSector SGPSector::FromShortString(const ST::string coordinates, INT8 h)
+{
+	Assert(coordinates.size() > 1);
+	UINT8 y = coordinates[0] - (coordinates[0] >= 'a' ? 'a' : 'A') + 1;
+	UINT8 x = coordinates.substr(1).to_int();
+	return SGPSector(x, y, h);
+}
+
+SGPSector SGPSector::FromSectorID(UINT32 s, INT8 h)
+{
+	UINT8 x = getXfromSectorID(s);
+	UINT8 y = getYfromSectorID(s);
+	return SGPSector(x, y, h);
+}
+
+bool SGPSector::operator==(const SGPSector& sector) const noexcept
+{
+	return x == sector.x && y == sector.y && z == sector.z;
+}
+
+bool SGPSector::operator!=(const SGPSector& sector) const noexcept
+{
+	return !(*this == sector);
+}
+
+bool SGPSector::operator<(const SGPSector& rhs) const noexcept
```

Comment:
I'm not sure about defining `<` for `SGPSector`. I think from a usage perspective it might not be clear what `sector1 < sector2` means.

Reply:
It seems quite intuitive to me, in two ways. I imagine the sector grid and basically anything that's:
- within the bigger rectangle formed by the current sector as one vertex and 1,1 the other is *less* than the current sector (ignoring equality here)
- to the left or to the top of the current sector is less